Sources/AnyPromise.swift

```diff
@@ -39,7 +39,7 @@ import Foundation
     @objc public func __catchOn(_ q: DispatchQueue, execute: @escaping (Any?) -> Any?) -> AnyPromise {
         return AnyPromise(__D: __AnyPromise(resolver: { resolve in
             self.__pipe { obj in
-                if obj is Error {
+                if obj is NSError || obj is Error {
```

Comment:
This is unnecessary.

Reply:
I think that swift has special handling of Error that can cause problems with reference counting.

Reply:
`NSError` is defined as `: Error` so this particular check surely cannot make a difference. If it does then 100% we need a comment to justify this existence of this check.